File: Backend-Deploy-Agent/src/models/dto/add_java_project_request_dto.py
from pydantic import BaseModel

class AddJavaProjectRequestDto(BaseModel):
    project_code: str
    project_name: str
    project_group: str
    git_repository: str
    docker_image_name: str
    docker_image_tag: str
    external_port: int
    internal_port: int
    network: str
    jdk_version: int
    host_project_path: str
    container_project_path: str

File: Frontend-Deploy-Center-UI/src/types/dto/AddJavaProjectRequestDto.ts
export interface AddJavaProjectRequestDto {
  project_code: string;
  project_name: string;
  project_group: string;
  git_repository?: string | null;
  docker_image_name: string;
  docker_image_tag: string;
  external_port: number;
  internal_port: number;
  network: string;
  jdk_version: number;
  host_project_path: string;
  container_project_path: string;
}
